libraries/ui/src/Nav.tsx

```diff
@@ -14,94 +13,152 @@ export type NavProps = React.PropsWithChildren<{
+  )}
+  >
+    <div className={clsx('nav-explore-section___dropdown-content flex flex-col gap-[14px] w-fit overflow-hidden text-pretty', innerClassName)}>
+      <h2 className="nav-explore-section__dropdown-title font-bold text-xl">Our courses</h2>
+      {courses?.map((course) => (
+        <a key={course.href} href={course.href} className="nav-explore-section__dropdown-link">
+          {course.isNew && (
+          <span className="nav-explore-section__new-badge text-bluedot-normal text-xl font-black pr-2">
+            New!
+          </span>
+          )}
+          {course.title}
+        </a>
+      ))}
+    </div>
+  </div>
+);
+
+const NavLinks: React.FC<{
+  exploreSectionInline?: boolean,
+  children: React.ReactNode,
+  courses: Array<{ title: string; href: string; isNew?: boolean }>,
+  exploreExpanded: boolean,
+  onToggleExplore: () => void,
+  className?: string
+}> = ({
+  exploreSectionInline, children, courses, exploreExpanded, onToggleExplore, className,
+}) => (
+  <div className={clsx('nav-links flex gap-9', className)}>
+    <div>
+      <button
+        type="button"
+        onClick={onToggleExplore}
+        className="nav-links__dropdown-button flex items-center gap-2 hover:text-bluedot-normal"
+      >
+        Explore
+        <DropdownIcon expanded={exploreExpanded} />
+      </button>
+      {exploreSectionInline && <ExploreSection expanded={exploreExpanded} courses={courses} className="nav-links__explore-section" innerClassName="pl-6 pt-6" />}
+    </div>
+    {children}
+  </div>
+);
+
+const LoginButtons: React.FC<{ className?: string }> = ({ className }) => (
+  <div className={clsx('nav__cta-container flex items-center', className)}>
+    <CTALinkOrButton className="nav__primary-cta" url="https://donate.stripe.com/5kA3fpgjpdJv6o89AA" isExternalUrl>Support us</CTALinkOrButton>
+    {/* <a className="nav__secondary-cta" href="https://course.aisafetyfundamentals.com/alignment?show=login" {...EXTERNAL_LINK_PROPS}>Log in</a>
+    <CTALinkOrButton className="nav__primary-cta" url="https://aisafetyfundamentals.com/" {...EXTERNAL_LINK_PROPS}>Get started for free</CTALinkOrButton> */}
+  </div>
+);
 
 export const Nav: React.FC<NavProps> = ({
   children, className, logo, courses,
 }) => {
-  const [isExpanded, setIsExpanded] = useState(false);
+  // Use a string rather than two booleans because `exploreExpanded = true`
+  // with `navExpanded = false` is not a valid state
+  const [expandedSections, setExpandedSections] = useState<'none' | 'nav' | 'nav-and-explore'>('none');
+  const navExpanded = expandedSections === 'nav' || expandedSections === 'nav-and-explore';
+  const exploreExpanded = expandedSections === 'nav-and-explore';
+
+  const onToggleExplore = useCallback(() => {
+    setExpandedSections((prev) => (prev === 'nav-and-explore' ? 'nav' : 'nav-and-explore'));
+  }, []);
+
+  const onToggleNav = useCallback(() => {
+    setExpandedSections((prev) => (prev === 'none' ? 'nav' : 'none'));
+  }, []);
```

Comment:
What do you think about something like this?
```suggestion
  const [isNavOpen, setIsNavOpen] = useState(false);
  const [isExploreOpen, setIsExploreOpen] = useState(false);

  const toggleNav = useCallback(() => {
    setIsNavOpen(prev => !prev);
    // Close explore dropdown when closing nav
    if (isNavOpen) {
      setIsExploreOpen(false);
    }
  }, [isNavOpen]);

  const toggleExplore = useCallback(() => {
    setIsExploreOpen(prev => !prev);
  }, []);
```

Reply:
Thanks for the suggestion, overall I prefer the single `useState` version because it makes impossible to set an invalid state. When it's this simple it doesn't make too much difference, but you could imagine e.g. opening and closing the nav via based on clicking elsewhere on the page, and we might lose track of maintaining the `isExploreOpen` state